server/middleware/setUpAuthentication.ts

```diff
@@ -32,6 +33,17 @@ export default function setUpAuth(): Router {
   router.use(async (req, res, next) => {
     await checkTokenValidityAndUpdate(req, res, next)
 
+    if (req.user) {
+      try {
```

Comment:
Just in case the parsing fails, we don't want it to make the entire app blow-up.